### DISCUSSION TITLE
Upgrade airflow in docker

### DISCUSSION THREAD

**Discussion by nasike** [apache/airflow#24632] [Q&A]
Hello, 

my organization is running airflow inside docker running in ec2 instance.
I am trying to upgrade airflow from v2.0 to 2.3,
But the steps to do this in docker is not clear to me.

what commands should I use to upgrade ?

should these command be run inside the docker containers or outside of the docker containers?

**Comment by jedcunningham** [comment]
You really need to rebuild your docker image with 2.3 in it instead of 2.0. Track down the Dockerfile your org is using, that's the best place to start.

**Reply by potiuk** [reply]
Yep. You should treat docker image as immutable set of software to run. Any change or installation you do inside the running container (based on this image) is temporary and is not going to survive container restart.

If you want to upgrade it, you simply need to get new image. And while doing it you should perform the db upgrade using the new image: https://airflow.apache.org/docs/apache-airflow/stable/installation/upgrading.html
